asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py

```diff
@@ -409,25 +410,30 @@ def create_fec_dataset(
              The planned FEC network which can be executed locally or submitted to alchemiscale.
         """
         # check that all ligands are unique in the series
-        if len(set(ligands)) != len(ligands):
-            count = Counter(ligands)
-            duplicated = [
-                key.compound_name for key, value in count.items() if value > 1
-            ]
-            raise ValueError(
-                f"ligand series contains {len(duplicated)} duplicate ligands: {duplicated}"
-            )
+        if ligands:
+            if len(set(ligands)) != len(ligands):
+                count = Counter(ligands)
+                duplicated = [
+                    key.compound_name for key, value in count.items() if value > 1
+                ]
+                raise ValueError(
+                    f"ligand series contains {len(duplicated)} duplicate ligands: {duplicated}"
+                )
 
-        # if any ligands lack a name, then raise an exception; important for
-        # ligands to have names for human-readable result gathering downstream
-        if missing := len([ligand for ligand in ligands if not ligand.compound_name]):
-            raise ValueError(
-                f"{missing} of {len(ligands)} ligands do not have names; names are required for ligands for downstream results handling"
-            )
+            # if any ligands lack a name, then raise an exception; important for
+            # ligands to have names for human-readable result gathering downstream
+            if missing := len(
+                [ligand for ligand in ligands if not ligand.compound_name]
+            ):
+                raise ValueError(
+                    f"{missing} of {len(ligands)} ligands do not have names; names are required for ligands for downstream results handling"
+                )
+        elif not graphml:
+            raise ValueError("No ligands or graphml provided to create the network.")
 
         # start by trying to plan the network
         planned_network = self.network_planner.generate_network(
-            ligands=ligands, central_ligand=central_ligand
+            ligands=ligands, central_ligand=central_ligand, graphml=graphml
```

Comment:
Do we still need to go through the network planner here? we could probably avoid adding the extra logic to `generate_network` this way and have a `from_graphml` method on the `PlannedNetwork` class which could have all of the graphml specific checks.